libraries/Microsoft.Bot.Builder.Prompts/NumberPrompt.cs

```diff
@@ -54,30 +56,24 @@ protected NumberPrompt(IModel model, PromptValidator<NumberResult<T>> validator
                 {
                     if (float.TryParse(result.Resolution["value"].ToString(), out float value))
                     {
-                        NumberResult<T> numberResult = new NumberResult<T>()
-                        {
-                            Value = (T)(object)value,
-                            Text = result.Text
-                        };
-                        if (await Validate(context, numberResult))
-                            return numberResult;
+                        numberResult.Status = RecognitionStatus.Recognized;
+                        numberResult.Value = (T)(object)value;
+                        numberResult.Text = result.Text;
+                        await Validate(context, numberResult);
                     }
                 }
                 else if (typeof(T) == typeof(int))
                 {
                     if (int.TryParse(result.Resolution["value"].ToString(), out int value))
                     {
-                        NumberResult<T> numberResult = new NumberResult<T>()
-                        {
-                            Value = (T)(object)value,
-                            Text = result.Text
-                        };
-                        if (await Validate(context, numberResult))
-                            return numberResult;
+                        numberResult.Status = RecognitionStatus.Recognized;
+                        numberResult.Value = (T)(object)value;
+                        numberResult.Text = result.Text;
+                        await Validate(context, numberResult);
                     }
                 }
             }
```

Comment:
Consider adding a final else that emits trace, or similar, saying "dunno what this thing is"